kolibri/plugins/user/assets/src/views/profile-page/index.vue

```diff
@@ -215,12 +217,27 @@
           if (!validateUsername(this.username)) {
             return this.$tr('usernameNotAlphaNumUnderscore');
           }
+          if (this.usernameAlreadyExists) {
+            return this.$tr('usernameAlreadyExists');
+          }
         }
         return '';
       },
       usernameIsInvalid() {
         return Boolean(this.usernameIsInvalidText);
       },
+      usernameAlreadyExists() {
+        return this.errorCode === 400;
```

Comment:
I think this might be an overly strong assumption, that any 400 is due to a duplicate username?

Reply:
@rtibbles I was kind of hesitant too about assuming that error 400 is caused by a duplicate username, but I also saw a definition here that uses this logic
https://github.com/learningequality/kolibri/blob/79c6a9046b153656c61c01b9375109d3ff395ae5/kolibri/plugins/user/assets/src/views/sign-up-page/index.vue#L192

Reply:
Let's use this logic for now, and create a new follow-up issue to improve the way that it's handled in both these cases.